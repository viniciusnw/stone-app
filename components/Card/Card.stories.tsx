import React from "react";
import { Card, StorybookWrapper } from "@Components";

const sample = {
  day: "25",
  month: "ABR",
  price: 10000,
  status: 'done',
  title: "Notebook LG 32GB - i7 / RTX 3070",
};

export const Default = {
  
};

export default {
  title: "Card",
  component: (props: any) => (
    <StorybookWrapper>
      <Card {...sample} />
    </StorybookWrapper>
  ),
};
