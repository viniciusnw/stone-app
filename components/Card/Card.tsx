import React from "react";
import { string } from "prop-types";
import styled from "styled-components/native";
import { View, Text, TouchableOpacity } from "react-native";
import Icon from "@Components/Icon";
import Label from "@Components/Label";

const StyledContainer = styled(View)`
  width: 100%;
  flex-direction: row;
  align-items: center;
  margin-bottom: ${({ theme }) => theme.spacing.regular};
`;

const StyledWrapper = styled(View)`
  flex-direction: row;
  justify-content: space-between;
  border: 1px solid ${({ theme }) => theme.color.border};
  flex: 1;
  padding: ${({ theme }) => theme.spacing.regular};
`;

const StyledInfo = styled(View)``;

const StyledTitle = styled(Text)`
  color: ${({ theme }) => theme.color.black};
  font-weight: 700;
  font-size: 16px;
  margin: 10px 0 5px;
`;

const StyledMore = styled(TouchableOpacity)``;

const StyledDateWrapper = styled(View)`
  align-items: center;
  margin-right: ${({ theme }) => theme.spacing.regular};
  padding-bottom: 5px;
`;

const StyledDay = styled(Text)`
  font-size: 42px;
  font-weight: 900;
  color: ${({ theme }) => theme.color.gray};
`;

const StyledMonth = styled(Text)`
  font-size: 22px;
  font-weight: 700;
  color: ${({ theme }) => theme.color.gray};
`;

const StyledHours = styled(Text)``;

type Props = {
  day: string;
  month: string;
  price: number;
  title: string;
  status: string
};

const Card: React.FC<Props> = ({ day, month, price, status, title }) => {
  function currencyFormat(num: number) {
    return "R$" + num.toFixed(2).replace(/(\d)(?=(\d{3})+(?!\d))/g, "$1,");
  }

  const parseStatus = (): { label: string; color: string } => {
    switch (status) {
      case "done":
        return { label: "Pago", color: "success" };
      case "pending":
        return { label: "Aguardando", color: "warning" };
      case "canceled":
        return { label: "Cancelado", color: "danger" };
      default:
        return { label: "Aguardando", color: "warning" };
    }
  };

  return (
    <StyledContainer>
      <StyledDateWrapper>
        <StyledDay>{day}</StyledDay>
        <StyledMonth>{month}</StyledMonth>
      </StyledDateWrapper>

      <StyledWrapper>
        <StyledInfo>
          <Label {...parseStatus()} />
          <StyledTitle>{title}</StyledTitle>
          <StyledHours>{currencyFormat(price)}</StyledHours>
        </StyledInfo>
        <StyledMore>
          <Icon icon="more" size={20} />
        </StyledMore>
      </StyledWrapper>
    </StyledContainer>
  );
};

export default Card;
