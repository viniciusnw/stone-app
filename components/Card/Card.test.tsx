import React from "react";
import renderer from "react-test-renderer";
import Card from "./Card";
import { darkTheme, lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import "jest-styled-components";

const sample = {
  day: "25",
  month: "ABR",
  price: 10000,
  status: "done",
  title: "Notebook LG 32GB - i7 / RTX 3070",
};

test("Card", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={darkTheme}>
        <Card {...sample} />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
