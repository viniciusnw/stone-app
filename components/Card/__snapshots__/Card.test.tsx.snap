// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 15,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "marginRight": 15,
        "paddingBottom": 5,
      }
    }
  >
    <Text
      style={
        {
          "color": "#666666",
          "fontSize": 42,
          "fontWeight": "900",
        }
      }
    >
      25
    </Text>
    <Text
      style={
        {
          "color": "#666666",
          "fontSize": 22,
          "fontWeight": "700",
        }
      }
    >
      ABR
    </Text>
  </View>
  <View
    style={
      {
        "borderColor": "#ddd",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-between",
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <View
      style={{}}
    >
      <View
        color="success"
        rounded={false}
        style={
          {
            "alignSelf": "flex-start",
            "backgroundColor": "green",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 0,
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 0,
            "flexDirection": "row",
            "paddingBottom": 5,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 5,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 12,
              "fontWeight": "700",
            }
          }
        >
          Pago
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#111111",
            "fontSize": 16,
            "fontWeight": "700",
            "marginBottom": 5,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 10,
          }
        }
      >
        Notebook LG 32GB - i7 / RTX 3070
      </Text>
      <Text
        style={{}}
      >
        R$10,000.00
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
